src/test/mir-opt/issue-72181/rustc.main.mir_map.0.mir

```diff
@@ -0,0 +1,385 @@
+// MIR for `main` 0 mir_map
+
+| User Type Annotations
+| 0: Canonical { max_universe: U0, variables: [CanonicalVarInfo { kind: Region(U0) }, CanonicalVarInfo { kind: Region(U0) }], value: TypeOf(DefId(2:6169 ~ core[2a38]::fmt[0]::{{impl}}[2]::new_v1[0]), UserSubsts { substs: [ReLateBound(DebruijnIndex(0), BrAnon(0))], user_self_ty: Some(UserSelfTy { impl_def_id: DefId(2:6167 ~ core[2a38]::fmt[0]::{{impl}}[2]), self_ty: std::fmt::Arguments<'_> }) }) } at $SRC_DIR/libstd/macros.rs:LL:COL
```

Comment:
So I think the symptom in CI is that this does a debug dump of various pieces of information that do not belong into the mir dump. I don't know why there's a difference between your system and CI, `DefId`s should be deterministic. Maybe it's stage1 vs stage2?

Anyway, I think we should just fix the symptom here, try removing the `println!` to reduce noise. If these `User Type Annotations` are still printed afterwards, maybe just stop printing the MIR for the `main` function entirely.

Please open an issue with your example and the command line and tag it for wg-mir-opt. We should stop debug dumping `DefId`s here.

Reply:
I've filed #73501 regarding this.

Reply:
Fixed this by removing the `println!`s in the test.